beater/pubsubbeat.go

```diff
@@ -114,6 +120,16 @@ func (bt *Pubsubbeat) Run(b *beat.Beat) error {
 			eventMap["attributes"] = m.Attributes
 		}
 
+		if m.Attributes["pubsubbeat.gzip"] == "true" {
```

Comment:
Maybe worth making this more generic so other compression methods like Snappy or Zstandard could be used.

Reply:
Good point! Changed to `pubsubbeat.compression` to make it more future proof.